File: Java/src/p_3_creational_patterns/p_9_builder/OrderBuilder.java
package p_3_creational_patterns.p_9_builder;

public abstract class OrderBuilder {

    public OrderBuilder() {
    }

    private Car orderedCar;
    public Car getOrderedCar(){
        if ( orderedCar == null)
            orderedCar = new Car();

        return orderedCar;
    }

    public abstract void setBrand ( String name);
    public abstract void setModel ( String name);
    public abstract void setColor ( String name);
    public abstract void setPower ( int power);

}

File: Net/03_creational_patterns/05_builder_pattern/OrderBuilder.cs
namespace _05_builder_pattern
{
    public abstract class OrderBuilder
    {
        public OrderBuilder()
        {
        }

        private Car _orderedCar;

        public Car GetOrderedCar()
        {
            if (_orderedCar == null)
                _orderedCar = new Car();

            return _orderedCar;
        }

        public abstract void SetBrand(string name);
        public abstract void SetModel(string name);
        public abstract void SetColor(string name);
        public abstract void SetPower(int power);
    }
}

